src/libasr/pass/intrinsic_functions.h

```diff
@@ -3088,21 +3088,17 @@ namespace Leadz {
         ASR::expr_t *two = b.i(2, arg_types[0]);
         ASR::expr_t* func_call_mod = b.CallIntrinsic(scope, {arg_types[0], arg_types[0]},
                                     {number, two}, return_type, 0, Mod::instantiate_Mod);
-        ASR::expr_t *if_cond = b.iLt(number, b.i(0, arg_types[0]));
-        ASR::expr_t *loop_cond = b.iGt(total_bits, b.i(0, arg_types[0]));
-
-        std::vector<ASR::stmt_t*> while_loop_body;
-        while_loop_body.push_back(b.If(b.iEq(func_call_mod, b.i(0, arg_types[0])), {
-            b.Assignment(result, b.i_tAdd(result, b.i(1, arg_types[0]), arg_types[0]))
-        }, {
-            b.Assignment(result, b.i(0, arg_types[0]))
-        }));
-        while_loop_body.push_back(b.Assignment(number, b.i_tDiv(number, two, arg_types[0])));
-        while_loop_body.push_back(b.Assignment(total_bits, b.i_tSub(total_bits, b.i(1, arg_types[0]), arg_types[0])));
-
         std::vector<ASR::stmt_t*> if_body; if_body.push_back(b.Assignment(result, b.i(0, arg_types[0])));
-        std::vector<ASR::stmt_t*> else_body; else_body.push_back(b.While(loop_cond, while_loop_body));
-        body.push_back(al, b.If(if_cond, if_body, else_body));
+        std::vector<ASR::stmt_t*> else_body; else_body.push_back(b.While(b.iGt(total_bits, b.i(0, arg_types[0])), {
+            b.If(b.iEq(func_call_mod, b.i(0, arg_types[0])), {
+                b.Assignment(result, b.i_tAdd(result, b.i(1, arg_types[0]), arg_types[0]))
+            }, {
+                b.Assignment(result, b.i(0, arg_types[0]))
+            }),
+            b.Assignment(number, b.i_tDiv(number, two, arg_types[0])),
+            b.Assignment(total_bits, b.i_tSub(total_bits, b.i(1, arg_types[0]), arg_types[0])),
+        }));
+        body.push_back(al, b.If(b.iLt(number, b.i(0, arg_types[0])), if_body, else_body));
```

Comment:
```suggestion
        body.push_back(al, b.If(b.iLt(number, b.i(0, arg_types[0])), {
            b.Assignment(result, b.i(0, arg_types[0]))
        }, {
            b.While(b.iGt(total_bits, b.i(0, arg_types[0])), {
                b.If(b.iEq(func_call_mod, b.i(0, arg_types[0])), {
                    b.Assignment(result, b.i_tAdd(result, b.i(1, arg_types[0]), arg_types[0]))
                }, {
                    b.Assignment(result, b.i(0, arg_types[0]))
                }),
                b.Assignment(number, b.i_tDiv(number, two, arg_types[0])),
                b.Assignment(total_bits, b.i_tSub(total_bits, b.i(1, arg_types[0]), arg_types[0])),
        })}));
```
See: `create_KMP_function` function, except `body.push_back(..)` keep everything inside `{..}`